enterprise/server/remote_execution/filecache/filecache.go

```diff
@@ -287,8 +287,20 @@ func (c *fileCache) FastLinkFile(ctx context.Context, node *repb.FileNode, outpu
 	}
 	start := time.Now()
 	if err := cloneOrLink(groupID, filecachePath(c.rootDir, key), outputPath); err != nil {
-		log.Warningf("Failed to link file from cache: %s", err)
-		return false
+		if status.IsAlreadyExistsError(err) {
+			// Remove and try again.
+			if err := os.RemoveAll(outputPath); err != nil {
+				log.Warningf("Failed to remove existing destination file: %s", err)
+				return false
+			}
+			if err := cloneOrLink(groupID, filecachePath(c.rootDir, key), outputPath); err != nil {
+				log.Warningf("Failed to link file from cache (after removing existing path): %s", err)
+				return false
+			}
+		} else {
+			log.Warningf("Failed to link file from cache: %s (%v)", err, err)
+			return false
+		}
```

Comment:
Does it make more sense to just always run os.RemoveAll before cloneOrLink here?